src/coreclr-debug/activate.ts

```diff
@@ -106,50 +106,63 @@ function showDotnetToolsWarning(message: string): void {
+    public static CORECLR_DEBUG_TYPE : string = "coreclr";
+    public static CLR_DEBUG_TYPE : string = "clr";
 
-// The default extension manifest calls this command as the adapterExecutableCommand
-// If the debugger components have not finished downloading, the proxy displays an error message to the user
-// Else it will launch the debug adapter
-export async function getAdapterExecutionCommand(platformInfo: PlatformInformation, eventStream: EventStream, packageJSON: any, extensionPath: string): Promise<AdapterExecutableCommand> {
-    let util = new CoreClrDebugUtil(common.getExtensionPath());
-
-    // Check for .debugger folder. Handle if it does not exist.
-    if (!CoreClrDebugUtil.existsSync(util.debugAdapterDir())) {
-        // our install.complete file does not exist yet, meaning we have not completed the installation. Try to figure out what if anything the package manager is doing
-        // the order in which files are dealt with is this:
-        // 1. install.Begin is created
-        // 2. install.Lock is created
-        // 3. install.Begin is deleted
-        // 4. install.complete is created
-
-        // install.Lock does not exist, need to wait for packages to finish downloading.
-        let installLock = false;
-        let debuggerPackage = getRuntimeDependencyPackageWithId("Debugger", packageJSON, platformInfo, extensionPath);
-        if (debuggerPackage && debuggerPackage.installPath)
-        {
-            installLock = await common.installFileExists(debuggerPackage.installPath, common.InstallFileType.Lock);
-        }
-        
-        if (!installLock) {
-            eventStream.post(new DebuggerNotInstalledFailure());
-            throw new Error('The C# extension is still downloading packages. Please see progress in the output window below.');
-        }
-        // install.complete does not exist, check dotnetCLI to see if we can complete.
-        else if (!CoreClrDebugUtil.existsSync(util.installCompleteFilePath())) {
-            let success: boolean = await completeDebuggerInstall(platformInfo, eventStream);
+    private platformInfo: PlatformInformation;
+    private eventStream: EventStream;
+    private packageJSON: any;
+    private extensionPath: string;
 
-            if (!success) {
-                eventStream.post(new DebuggerNotInstalledFailure());
-                throw new Error('Failed to complete the installation of the C# extension. Please see the error in the output window below.');
+    constructor(platformInfo: PlatformInformation, eventStream: EventStream, packageJSON: any, extensionPath: string) {
+        this.platformInfo = platformInfo;
+        this.eventStream = eventStream;
+        this.packageJSON = packageJSON;
+        this.extensionPath = extensionPath;
+    }
+
+    createDebugAdapterDescriptor(session: vscode.DebugSession, executable: vscode.DebugAdapterExecutable | undefined): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
```

Comment:
>| undefined [](start = 105, length = 11)

why this?

Reply:
Took defn from https://github.com/Microsoft/vscode/blob/3e560c1b1ed9c58d7d3871f048b035474f4f1e85/src/vs/vscode.d.ts#L8489

Reply:
I think you can remove it. The example here doesn't have it: [https://code.visualstudio.com/updates/v1_30](https://code.visualstudio.com/updates/v1_30)

---
In reply to: [241589154](https://github.com/OmniSharp/omnisharp-vscode/pull/2746#discussion_r241589154) [](ancestors = 241589154)

Reply:
Technically, since we don't have any executable defined in the package.json. It will be undefined every time.

Reply:
oic.

---
In reply to: [241590911](https://github.com/OmniSharp/omnisharp-vscode/pull/2746#discussion_r241590911) [](ancestors = 241590911)